app/models/environment.rb

```diff
@@ -0,0 +1,10 @@
+class Environment < ApplicationRecord
+
+  has_many :tokens
```

Comment:
`dependent: :nullify` to set the `environment_id` on dependent records to nil. Will have to ensure that token and access request template (index, show etc) handle displaying of nilled values.